src/plugins/vis_type_xy/public/xy_vis_fn.ts

```diff
@@ -48,15 +43,217 @@ export const createVisTypeXyVisFn = (): VisTypeXyExpressionFunctionDefinition =>
+        defaultMessage: 'Show tooltip on hover',
+      }),
+    },
+    legendPosition: {
+      types: ['string'],
+      help: i18n.translate('visTypeXy.function.args.legendPosition.help', {
+        defaultMessage: 'Position the legend on top, bottom, left, right of the chart',
+      }),
+    },
+    categoryAxes: {
+      types: ['category_axis'],
+      help: i18n.translate('visTypeXy.function.args.categoryAxes.help', {
+        defaultMessage: 'Category axis config',
+      }),
+      multi: true,
+    },
+    thresholdLine: {
+      types: ['threshold_line'],
+      help: i18n.translate('visTypeXy.function.args.thresholdLine.help', {
+        defaultMessage: 'Threshold line config',
+      }),
+    },
+    labels: {
+      types: ['label'],
+      help: i18n.translate('visTypeXy.function.args.labels.help', {
+        defaultMessage: 'Chart labels config',
+      }),
+    },
+    orderBucketsBySum: {
+      types: ['boolean'],
+      help: i18n.translate('visTypeXy.function.args.orderBucketsBySum.help', {
+        defaultMessage: 'Order buckets by sum',
+      }),
+    },
+    seriesParams: {
+      types: ['series_param'],
+      help: i18n.translate('visTypeXy.function.args.seriesParams.help', {
+        defaultMessage: 'Series param config',
+      }),
+      multi: true,
+    },
+    valueAxes: {
+      types: ['value_axis'],
+      help: i18n.translate('visTypeXy.function.args.valueAxes.help', {
+        defaultMessage: 'Value axis config',
+      }),
+      multi: true,
+    },
+    radiusRatio: {
+      types: ['number'],
+      help: i18n.translate('visTypeXy.function.args.radiusRatio.help', {
+        defaultMessage: 'Dot size ratio',
+      }),
+    },
+    gridCategoryLines: {
+      types: ['boolean'],
+      help: i18n.translate('visTypeXy.function.args.gridCategoryLines.help', {
+        defaultMessage: 'Show grid category lines in chart',
+      }),
+    },
+    gridValueAxis: {
       types: ['string'],
-      default: '"{}"',
-      help: 'xy vis config',
+      help: i18n.translate('visTypeXy.function.args.gridValueAxis.help', {
+        defaultMessage: 'Name of value axis for which we show grid',
+      }),
+    },
+    isVislibVis: {
+      types: ['boolean'],
+      help: i18n.translate('visTypeXy.function.args.isVislibVis.help', {
+        defaultMessage:
+          'Flag to indicate old vislib visualizations. Used for backwards compatibility including colors',
+      }),
+    },
+    detailedTooltip: {
+      types: ['boolean'],
+      help: i18n.translate('visTypeXy.function.args.detailedTooltip.help', {
+        defaultMessage: 'Show detailed tooltip',
+      }),
+    },
+    fittingFunction: {
+      types: ['string'],
+      help: i18n.translate('visTypeXy.function.args.fittingFunction.help', {
+        defaultMessage: 'Name of fitting function',
+      }),
+    },
+    times: {
+      types: ['time_marker'],
+      help: i18n.translate('visTypeXy.function.args.times.help', {
+        defaultMessage: 'times',
```

Comment:
Can we add a more detailed description here?

Reply:
done